pypesto/visualize/sampling.py

```diff
@@ -92,119 +112,347 @@ def sampling_fval_trace(
+        n_variables = len(condition_ids)
+        y_names = condition_ids
+        n_subplots = len(observable_ids)
     else:
-        fig = ax.get_figure()
+        raise ValueError(f'Unsupported groupby value: {groupby}')
+
+    percentiles_opacity = sorted(
+        # min 30%, max 100%, opacity
+        np.linspace(0.3 * RGBA_MAX, RGBA_MAX, len(percentiles)),
+        reverse=True,
+    )
+    if axes is not None:
+        fig = axes.get_figure()
+        if not isinstance(axes, np.ndarray):
+            axes = np.array([[axes]])
+    cmap = plt.cm.viridis
+    cmap_min = RGBA_MIN
+    cmap_max = 0.85*(RGBA_MAX - RGBA_MIN) + RGBA_MIN  # exclude yellows
 
-    axes = dict(zip(ynames, ax.flat))
-
-    for i, level in enumerate(alpha_sorted):
-
-        # Get upper and lower bounds for the confidence level
-        lb, ub = calculate_prediction_profiles(values, alpha=level / 100)
-
-        # Get the median
-        _median = np.percentile(values, 50, axis=1)
-
-        n_timepoints = _median.shape[1]
-        n_conditions = _median.shape[0]
-
-        # Loop over observables/states
-        for j, iplot in enumerate(ynames):
-            # Create array for X axis
-            t = np.arange(n_timepoints)
-            # Loop over experimental conditions
-            for k in range(n_conditions):
-                # Plot confidence region
-                axes[iplot].fill_between(t,
-                                         ub[k, :, j],
-                                         lb[k, :, j],
-                                         facecolor=colors[i],
-                                         alpha=0.9,
-                                         label=str(level) + '% CI')
-                # Plot median
-                axes[iplot].plot(t, _median[k, :, j],
-                                 'k-', label='MCMC median')
-                axes[iplot].set_ylabel(yname + '_' + str(j + 1))
-                # Update X axis array
-                t += n_timepoints
+    # define colormap
+    variables_color = [
+        list(cmap(v))[:LEN_RGB]
+        for v in np.linspace(cmap_min, cmap_max, n_variables)
+    ]
+
+    if axes is None:
+        n_row = int(np.round(np.sqrt(n_subplots)))
+        n_col = int(np.ceil(n_subplots / n_row))
+        fig, axes = plt.subplots(n_row, n_col, figsize=size, squeeze=False)
+    elif len(axes.flat) < n_subplots:
+        raise ValueError(
+            'Provided `ax` contains insufficient subplots. At least '
+            f'{n_subplots} are required.'
+        )
+
+    def get_statistic_data(
+            statistic: str,
+            condition_index: int,
+            observable_id: int,
+    ):
+        condition_result = summary[statistic].conditions[condition_index]
+        t = condition_result.timepoints
+        observable_index = condition_result.observable_ids.index(observable_id)
+        y = condition_result.output[:, observable_index]
+        return (t, y)
+
+    if groupby == CONDITION:
```

Comment:
I think these might want to become own private functions ... makes dependencies clearer ...